### PR TITLE
zfs: try pool import

### LINKED ISSUE #3976: LXD failing to start container when using ZFS pool on other hard drive, only after reboot [edge]
# Required information
 * Distribution: debian 9 strech
 * Kernel version: 4.9.0-4-amd64  
 * LXD version: edge (latest; checked)
 * Storage backend in use: zfs

Using ZFS from sda1. Configured storage on another HDD when using lxd init.

# Issue description

Container failing to start with normal start command.

No custom GPU forwarding or Authority configuration, fresh install using -- edge with snapd.

# Steps to reproduce
1. Configure LXD to use ZFS from other hard drive using lxd init (use existing device -- yes)
2. launch container debian - strech (container works)
3. reboot host
2. launch using lxc start build-bot
3. lxc start build-bot
4. error: 
error: no such file or directory
Try `lxc info --show-log build-bot` for more info

# Information to attach
**Container log**
Name: build-bot
Remote: unix://
Architecture: x86_64
Created: 2017/10/26 01:06 UTC
Status: Stopped
Type: persistent
Profiles: default

**Container configuration**
 - [ ] Container configuration (`lxc config show NAME --expanded`)
architecture: x86_64
config:
  image.architecture: amd64
  image.description: Debian stretch amd64 (20171025_22:42)
  image.os: Debian
  image.release: stretch
  image.serial: "20171025_22:42"
  volatile.base_image: 21fae0325d00b6077fac9b399ecb9a5b59c88da4f96750ee61da2b9842339d43
  volatile.eth0.hwaddr: 00:16:3e:d4:b2:45
  volatile.eth0.name: eth0
  volatile.idmap.base: "0"
  volatile.idmap.next: '[{"Isuid":true,"Isgid":true,"Hostid":1000000,"Nsid":0,"Maprange":1000000000}]'
  volatile.last_state.idmap: '[{"Isuid":true,"Isgid":true,"Hostid":1000000,"Nsid":0,"Maprange":1000000000}]'
  volatile.last_state.power: RUNNING
devices:
  eth0:
    nictype: bridged
    parent: lxdbr0
    type: nic
  root:
    path: /
    pool: default
    type: disk
ephemeral: false
profiles:
- default
stateful: false
description: ""
**Main daemon log**
lvl=info msg="LXD 2.19 is starting in normal mode" path=/var/snap/lxd/common/lxd t=2017-10-25T22:58:38+0000
lvl=warn msg="AppArmor support has been disabled because of lack of kernel support" t=2017-10-25T22:58:38+0000
lvl=warn msg="CGroup memory swap accounting is disabled, swap limits will be ignored." t=2017-10-25T22:58:38+0000
lvl=info msg="Kernel uid/gid map:" t=2017-10-25T22:58:38+0000
lvl=info msg=" - u 0 0 4294967295" t=2017-10-25T22:58:38+0000
lvl=info msg=" - g 0 0 4294967295" t=2017-10-25T22:58:38+0000
lvl=info msg="Configured LXD uid/gid map:" t=2017-10-25T22:58:38+0000
lvl=info msg=" - u 0 1000000 1000000000" t=2017-10-25T22:58:38+0000
lvl=info msg=" - g 0 1000000 1000000000" t=2017-10-25T22:58:38+0000
lvl=info msg="Expiring log files" t=2017-10-25T22:58:38+0000
lvl=info msg="Done expiring log files" t=2017-10-25T22:58:38+0000
lvl=info msg="Starting /dev/lxd handler" t=2017-10-25T22:58:38+0000
lvl=info msg="LXD isn't socket activated" t=2017-10-25T22:58:40+0000
lvl=info msg="REST API daemon:" t=2017-10-25T22:58:40+0000
lvl=info msg=" - binding Unix socket" socket=/var/snap/lxd/common/lxd/unix.socket t=2017-10-25T22:58:40+0000
lvl=info msg="Pruning expired images" t=2017-10-25T22:58:40+0000
lvl=info msg="Done pruning expired images" t=2017-10-25T22:58:40+0000
lvl=info msg="Updating images" t=2017-10-25T22:58:40+0000
lvl=info msg="Updating instance types" t=2017-10-25T22:58:40+0000
lvl=info msg="Done updating images" t=2017-10-25T22:58:40+0000
lvl=info msg="Done updating instance types" t=2017-10-25T22:58:49+0000
lvl=info msg="Creating ZFS storage pool \"default\"." t=2017-10-26T00:58:53+0000
lvl=info msg="Created ZFS storage pool \"default\"." t=2017-10-26T00:58:54+0000
alias=debian/stretch image=21fae0325d00b6077fac9b399ecb9a5b59c88da4f96750ee61da2b9842339d43 lvl=info msg="Downloading image" operation=c7761004-fc13-425f-a51e-61493369e0fd server=https://images.linuxcontainers.org t=2017-10-26T01:01:57+0000 trigger=/1.0/operations/c7761004-fc13-425f-a51e-61493369e0fd
alias=debian/stretch image=21fae0325d00b6077fac9b399ecb9a5b59c88da4f96750ee61da2b9842339d43 lvl=info msg="Image downloaded" operation=c7761004-fc13-425f-a51e-61493369e0fd server=https://images.linuxcontainers.org t=2017-10-26T01:06:58+0000 trigger=/1.0/operations/c7761004-fc13-425f-a51e-61493369e0fd
ephemeral=false lvl=info msg="Creating container" name=build-bot t=2017-10-26T01:06:58+0000
ephemeral=false lvl=info msg="Created container" name=build-bot t=2017-10-26T01:06:58+0000
lvl=warn msg="Unable to update backup.yaml at this time." name=build-bot t=2017-10-26T01:07:02+0000
action=start created=2017-10-26T01:06:58+0000 ephemeral=false lvl=info msg="Starting container" name=build-bot stateful=false t=2017-10-26T01:07:02+0000 used=1970-01-01T00:00:00+0000
action=start created=2017-10-26T01:06:58+0000 ephemeral=false lvl=info msg="Started container" name=build-bot stateful=false t=2017-10-26T01:07:02+0000 used=1970-01-01T00:00:00+0000
lvl=info msg="Asked to shutdown by API, shutting down containers." t=2017-10-26T01:34:02+0000
action=shutdown created=2017-10-26T01:06:58+0000 ephemeral=false lvl=info msg="Shutting down container" name=build-bot t=2017-10-26T01:34:02+0000 timeout=30s used=2017-10-26T01:07:02+0000
action=shutdown created=2017-10-26T01:06:58+0000 ephemeral=false lvl=info msg="Shut down container" name=build-bot t=2017-10-26T01:34:04+0000 timeout=30s used=2017-10-26T01:07:02+0000
lvl=warn msg="Unable to update backup.yaml at this time." name=build-bot t=2017-10-26T01:34:04+0000
lvl=info msg="Stopping REST API handler:" t=2017-10-26T01:34:04+0000
lvl=info msg=" - closing socket" socket=[::]:8443 t=2017-10-26T01:34:04+0000
lvl=info msg=" - closing socket" socket=/var/snap/lxd/common/lxd/unix.socket t=2017-10-26T01:34:04+0000
lvl=info msg="Stopping /dev/lxd handler" t=2017-10-26T01:34:04+0000
lvl=info msg="Stopped /dev/lxd handler" t=2017-10-26T01:34:04+0000
lvl=info msg="Unmounting temporary filesystems" t=2017-10-26T01:34:04+0000
lvl=info msg="Done unmounting temporary filesystems" t=2017-10-26T01:34:04+0000
lvl=info msg="Closing the database" t=2017-10-26T01:34:04+0000
lvl=info msg="Saving simplestreams cache" t=2017-10-26T01:34:04+0000
lvl=info msg="Saved simplestreams cache" t=2017-10-26T01:34:04+0000
lvl=info msg="LXD 2.19 is starting in normal mode" path=/var/snap/lxd/common/lxd t=2017-10-26T11:11:37+0000
lvl=warn msg="AppArmor support has been disabled because of lack of kernel support" t=2017-10-26T11:11:37+0000
lvl=warn msg="CGroup memory swap accounting is disabled, swap limits will be ignored." t=2017-10-26T11:11:37+0000
lvl=info msg="Kernel uid/gid map:" t=2017-10-26T11:11:37+0000
lvl=info msg=" - u 0 0 4294967295" t=2017-10-26T11:11:37+0000
lvl=info msg=" - g 0 0 4294967295" t=2017-10-26T11:11:37+0000
lvl=info msg="Configured LXD uid/gid map:" t=2017-10-26T11:11:37+0000
lvl=info msg=" - u 0 1000000 1000000000" t=2017-10-26T11:11:37+0000
lvl=info msg=" - g 0 1000000 1000000000" t=2017-10-26T11:11:37+0000
lvl=info msg="Expiring log files" t=2017-10-26T11:11:37+0000
lvl=info msg="Done expiring log files" t=2017-10-26T11:11:37+0000
lvl=info msg="Starting /dev/lxd handler" t=2017-10-26T11:11:37+0000
lvl=info msg="LXD isn't socket activated" t=2017-10-26T11:11:37+0000
lvl=info msg="REST API daemon:" t=2017-10-26T11:11:37+0000
lvl=info msg=" - binding Unix socket" socket=/var/snap/lxd/common/lxd/unix.socket t=2017-10-26T11:11:37+0000
lvl=info msg=" - binding TCP socket" socket=[::]:8443 t=2017-10-26T11:11:37+0000
lvl=info msg="Pruning expired images" t=2017-10-26T11:11:37+0000
lvl=info msg="Done pruning expired images" t=2017-10-26T11:11:37+0000
lvl=info msg="Updating images" t=2017-10-26T11:11:37+0000
lvl=info msg="Updating instance types" t=2017-10-26T11:11:37+0000
lvl=info msg="Done updating images" t=2017-10-26T11:11:39+0000
lvl=info msg="Done updating instance types" t=2017-10-26T11:11:40+0000
lvl=eror msg="Failed to mount ZFS dataset \"default/containers/build-bot\" onto \"/var/snap/lxd/common/lxd/storage-pools/default/containers/build-bot\"." t=2017-10-26T11:11:47+0000
lvl=eror msg="Failed to mount ZFS dataset \"default/containers/build-bot\" onto \"/var/snap/lxd/common/lxd/storage-pools/default/containers/build-bot\"." t=2017-10-26T11:18:09+0000
lvl=eror msg="Failed to mount ZFS dataset \"default/containers/build-bot\" onto \"/var/snap/lxd/common/lxd/storage-pools/default/containers/build-bot\"." t=2017-10-26T11:22:14+0000

### PR DESCRIPTION
**Pull request by brauner**
Closes #3976.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

**Comment by stgraber** [comment]
@brauner is poolName the actual zpool name or the dataset?

We need to do "zpool import tank" not "zpool import tank/lxd", so if it's the dataset, we should split on / and take [0].

**Comment by stgraber** [comment]
@brauner ping?

**Comment by brauner** [comment]
Yeah, will take a look once I get some free cycles a the conference.

**Comment by brauner** [comment]
@stgraber, updated.

**Comment by stgraber** [comment]
we'll want to re-try this after my branch is merged, should get us a cleaner test run.

**Comment by stgraber** [comment]
jenkins: test this please

File: lxd/storage_zfs.go [hunk 1]
```diff
@@ -87,24 +87,32 @@ func (s *storageZfs) StoragePoolCheck() error {
...
-			"import",
-			"-d", disksPath, poolName)
-		if err != nil {
-			return fmt.Errorf("ZFS storage pool \"%s\" could not be imported: %s", poolName, output)
+		msg, err = shared.RunCommand("zpool", "import", "-d", disksPath, poolName)
+	} else {
+		purePoolName := poolName
+		idx := strings.Index(poolName, "/")
+		if idx != -1 {
+			purePoolName = poolName[:idx]
```
Comment (stgraber) [inline lxd/storage_zfs.go]: That looks more complicated than it should be, don't you just want: strings.Split(poolName, "/")[0] here?

Comment (stgraber) [reply lxd/storage_zfs.go]: @brauner saw that one?

Comment (brauner) [reply lxd/storage_zfs.go]: Oh weird, the connection must have broken off when I pushed this on the plane.

Comment (brauner) [reply lxd/storage_zfs.go]: done

**Pull request merged**